src/goose/toolkit/web_browser.py

```diff
@@ -0,0 +1,442 @@
+            prog_id, _ = winreg.QueryValueEx(key, "ProgId")
+
+        with winreg.OpenKey(winreg.HKEY_CLASSES_ROOT, f"{prog_id}\\shell\\open\\command") as cmd_key:
+            command, _ = winreg.QueryValueEx(cmd_key, None)
+
+        if command.startswith('"'):
+            executable = command.split('"')[1]
+        else:
+            executable = command.split(" ")[0]
+
+        return os.path.basename(executable)
+
+    except Exception as e:
+        print(f"Error retrieving default browser on Windows: {e}")
+        return None
+
+
+def get_default_browser_macos() -> str:
+    try:
+        import plistlib
+        import os
+
+        plist_path = os.path.expanduser(
+            "~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure.plist"
+        )
+
+        if not os.path.exists(plist_path):
+            print(f"Launch services plist not found at: {plist_path}")
+            return None
+
+        with open(plist_path, "rb") as fp:
+            plist = plistlib.load(fp)
+            handlers = plist.get("LSHandlers", [])
+
+            for handler in handlers:
+                scheme = handler.get("LSHandlerURLScheme")
+                if scheme and scheme.lower() == "http":
+                    return handler.get("LSHandlerRoleAll")
+
+        return None
+    except Exception as e:
+        print(f"Error retrieving default browser on macOS: {e}")
+        return None
+
+
+def get_default_browser_linux() -> str:
+    try:
+        result = subprocess.run(
+            ["xdg-settings", "get", "default-web-browser"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
+        )
+
+        if result.returncode != 0:
+            print(f"Error: {result.stderr.strip()}")
+            return None
+
+        desktop_file = result.stdout.strip()
+        desktop_paths = [
+            os.path.expanduser("~/.local/share/applications/"),
+            "/usr/share/applications/",
+            "/usr/local/share/applications/",
+        ]
+
+        for path in desktop_paths:
+            desktop_file_path = os.path.join(path, desktop_file)
+            if os.path.exists(desktop_file_path):
+                with open(desktop_file_path, "r") as f:
+                    for line in f:
+                        if line.startswith("Name="):
+                            name = line.split("=", 1)[1].strip()
+                            return name
+        return desktop_file.replace(".desktop", "")
+
+    except Exception as e:
+        print(f"Error retrieving default browser on Linux: {e}")
+        return None
+
+
+def get_default_browser() -> str:
+    if sys.platform.startswith("darwin"):
+        return get_default_browser_macos()
+    # other platforms are not enabled yet.
+    # elif sys.platform.startswith("win"):
+    #     return get_default_browser_windows()
+    # elif sys.platform.startswith("linux"):
+    #     return get_default_browser_linux()
+    else:
+        print("Unsupported operating system.")
```

Comment:
if you wanted, we could print the platform `print(f"Unsupported platform {sys.platform}")`